parsers/maps/mapcanvas.py

```diff
@@ -384,3 +419,187 @@ def contextMenuEvent(self, event):
+        filename = "{zone}_{recording}.txt".format(
+            zone=zone_key,
+            recording=clean_name)
+
+        # Make sure the directory exists
+        record_dir = MAP_FILES_PATHLIB.joinpath('recordings')
+        if not os.path.exists(record_dir):
+            try:
+                print("Creating custom map directory.")
+                os.makedirs(record_dir)
+            except Exception as e:
+                print("Failed to make custom map directory: %s" % e)
+        return record_dir.joinpath(filename)
+
+    def start_path_recording(self, name=None):
+        print("Start recording!")
+        if self._path_recording:
+            return
+
+        if name:
+            path_name = name
+            ok_pressed = True
+        else:
+            path_name, ok_pressed = QInputDialog.getText(
+                self,  # parent
+                "Start Recording Path",  # title
+                "Name of path to record:",  # label
+                echo=QLineEdit.Normal,
+                text="")
+        if ok_pressed:
+            self._path_recording_name = path_name
+            try:
+                self._path_file = open(self._get_path_filename(), 'a')
+                self._path_recording = True
+                self._path_last_loc = None
+            except Exception as e:
+                print("Failed to open pathfile: %s" % e)
+
+    def rename_path_recording(self, new_name=None):
+        print("Rename recording!")
+        if not self._path_recording:
+            return
+
+        if new_name:
+            path_name = new_name
+            ok_pressed = True
+        else:
+            path_name, ok_pressed = QInputDialog.getText(
+                self,  # parent
+                "Rename Path",  # title
+                "New path name:",  # label
+                echo=QLineEdit.Normal,
+                text=self._path_recording_name)
+
+        if ok_pressed:
+            old_path_name = self._path_recording_name
+            new_path_name = path_name
+            try:
+                self._path_file.close()
+                self._path_file = None
+            except Exception as e:
+                print("Failed to close path recording file: %s" % e)
+                return
+            try:
+                os.rename(self._get_path_filename(custom_name=old_path_name),
+                          self._get_path_filename(custom_name=new_path_name))
+                self._path_recording_name = new_path_name
+            except Exception as e:
+                print("Failed to rename path recording file: %s" % e)
+                self._path_recording = False
+                return
+            try:
+                self._path_file = open(self._get_path_filename(), 'a')
+            except Exception as e:
+                print("Failed to open renamed path recording file: %s" % e)
+                self._path_recording = False
+                return
+
+        # This is ... broken
```

Comment:
TODO: I was trying to use the real validators for the filename input, but I will need to remake the whole panel manually. This isn't hard, I just didn't get to it. Needs cleanup.